ui/src/app/pages/data/data-set/export-dataset-modal.tsx

```diff
@@ -0,0 +1,252 @@
+import {BillingStatus, DataSet, DataSetRequest, KernelTypeEnum, } from 'generated/fetch';
+import * as fp from 'lodash/fp';
+import * as React from 'react';
+import {useEffect, useState} from 'react';
+
+import {FlexRow} from 'app/components/flex';
+import {SmallHeader, styles as headerStyles} from 'app/components/headers';
+import {RadioButton, Select, TextInput} from 'app/components/inputs';
+import {ErrorMessage} from 'app/components/messages';
+import {Spinner} from 'app/components/spinners';
+import colors from 'app/styles/colors';
+import {AnalyticsTracker} from 'app/utils/analytics';
+import {ACTION_DISABLED_INVALID_BILLING} from 'app/utils/strings';
+import {WorkspaceData} from 'app/utils/workspace-data';
+import {WorkspacePermissionsUtil} from 'app/utils/workspace-permissions';
+import {validate} from 'validate.js';
+
+interface Props {
+  closeFunction: Function;
+  dataset: DataSet;
+}
+
+// HocProps includes all props that are inherited through HOC
+interface HocProps extends Props {
+  workspace: WorkspaceData;
+}
+
+export const ExportDatasetModal: (props: Props) => JSX.Element = fp.flow(withCurrentWorkspace())(
+  ({workspace, dataset, closeFunction}: HocProps) => {
+    const [existingNotebooks, setExistingNotebooks] = useState(undefined);
+    const [kernelType, setKernelType] = useState(KernelTypeEnum.Python);
+    const [isExporting, setIsExporting] = useState(false); // replace w/ undefined notebooks list? // test case - no notebooks in workspace
+    const [creatingNewNotebook, setCreatingNewNotebook] = useState(true);
+    const [notebookName, setNotebookName] = useState('');
+    const [codePreview, setCodePreview] = useState(null);
+    const [loadingNotebook, setIsLoadingNotebook] = useState(false);
+    const [errorMsg, setErrorMsg] = useState(null);
+
+    async function exportDataset() {
+      AnalyticsTracker.DatasetBuilder.Export(kernelType);
+
+      setErrorMsg(null);
+      setIsExporting(true);
+      try {
+        await dataSetApi().exportToNotebook(
+          workspace.namespace, workspace.id,
+          {
+            dataSetRequest: createDataSetRequest(),
+            kernelType: kernelType,
+            notebookName: notebookName,
+            newNotebook: creatingNewNotebook
+          });
+        // Open notebook in a new tab and return back to the Data tab
+        const notebookUrl = `/workspaces/${workspace.namespace}/${workspace.id}/notebooks/preview/` +
+          appendNotebookFileSuffix(encodeURIComponentStrict(notebookName));
+        navigateByUrl(notebookUrl);
+      } catch (e) {
+        console.error(e);
+        setIsExporting(false);
+        setErrorMsg('The request cannot be completed. Please try again or contact Support in the left hand navigation');
+      }
+    }
+
+    function createDataSetRequest(): DataSetRequest {
+      return {
+        name: dataset ? dataset.name : 'dataset',
+        ...(dataset.id ? {
```

Comment:
nit: not sure if we have an official stance on styling for multi-line ternaries but I find this a lot harder to read than:

```
thing
    ? { [things] }
    : {
        [things]
    }

Reply:
sure! I like that